src/components/Sidebar.jsx

```diff
@@ -1,149 +1,100 @@
-            (discount) => (
-              <FormControlLabel
-                key={discount}
-                control={
-                  <Checkbox
-                    name={discount}
-                    onChange={(e) => handleCheckboxChange(e, "discount")}
-                  />
-                }
-                label={discount}
-              />
-            )
-          )}
-        </FormGroup>
-      </div>
+//   // TODO: create function which combines all queries: brand, price, category...
 
-      <div className="filter-section">
-        <Typography variant="h6">Rating</Typography>
-        <FormGroup>
-          {["5", "4", "3", "2", "1"].map((rating) => (
-            <FormControlLabel
-              key={rating}
-              control={
-                <Checkbox
-                  name={rating}
-                  onChange={(e) => handleCheckboxChange(e, "rating")}
-                />
-              }
-              label={`${"★".repeat(rating)}${"☆".repeat(5 - rating)}`}
-            />
-          ))}
-        </FormGroup>
-      </div>
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { Checkbox, FormControlLabel, FormGroup, Typography } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+
+const Sidebar = ({ setFilters }) => {
+  const [brands, setBrands] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function fetchFilters() {
+      try {
+        const response = await fetch("http://localhost:3000/products");
+        const data = await response.json();
+        const brands = [...new Set(data.map((product) => product.brand))];
+        setBrands(brands.map((brand) => ({ name: brand })));
+        const categories = [...new Set(data.flatMap((product) => product.category))];
+        setCategories(categories.map((category) => ({ name: category })));
+      } catch (error) {
+        console.error("Failed to fetch data:", error);
+      }
+    }
 
-      <div className="filter-section">
-        <Typography variant="h6">Categories</Typography>
-        <FormGroup>
-          {categories.map((category) => (
-            <FormControlLabel
-              key={category}
-              control={
-                <Checkbox
-                  name={category}
-                  onChange={(e) => handleCheckboxChange(e, "categories")}
+    fetchFilters();
+  }, []);
+
+  const handleCheckboxChange = (event, category) => {
+    setFilters((prevFilters) => {
+      const newFilters = {
+        ...prevFilters,
+        [category]: event.target.checked
+            ? [...(prevFilters[category] || []), event.target.name]
+            : (prevFilters[category] || []).filter((item) => item !== event.target.name),
+      };
+
+      const params = new URLSearchParams(newFilters).toString();
+      navigate(`?${params}`);
+
+      return newFilters;
+    });
+  };
```

Comment:
Category filter is not working. When list of selected brands is empty, url should not include brand. I.e. currently: `http://localhost:5173/?brand=&categories=` should be `http://localhost:5173`